docs/source/guides/upload.md

```diff
@@ -371,11 +376,89 @@ In addition to [`upload_file`] and [`upload_folder`], the following functions al
+in advance to be aware of it. It's also better to get assistance if you have questions on your process.
+- **Number of files**: There are two practical limits about the number of files in your repo. Under the hood, the Hub uses
+Git to version the data which has its limitations:
+    - The total number of files in the repo. It cannot exceed 1M files. If you are planning to upload more than 1M files,
+      we recommend to merge the data into fewer files. For example, json files can be merged into a single jsonl file or
+      large datasets can be exported as parquet files.
+    - The maximum number of files per folder. It cannot exceed 10k files per folder. A simple solution for that is to
+      create a repository structure that uses subdirectories. A repo with 1k folder from `000/` to `999/` each one of
+      them containing at most 1000 files is already enough.
+- **File size**: individual files also have a limit on their maximum size. In the case of uploading large files (e.g.
+model weights), we strongly recommend to split them **into chunks of around 10GB each**. There are a few reasons for this:
+    - Uploading and downloading smaller files is easier both for you and the other users. Connection issues can always
+      happen when streaming data and smaller files avoid resuming from the beginning in case of errors.
+    - Files are served to the users using CloudFront. From our experience, files above 30GB are not cached by this service
+      leading to a slower download speed.
+    - A hard-limit of 50GB has been set on the Hub. If you try to commit larger files, an error will be returned by the server.
+- **Number of commits**: The total number of commits on your repo history. There is no hard limit for this. However, from
+our experience, the user experience on the Hub start to degrade after a few thousands commits. We are always working to
+improve the service but one must always remember that a git repository is not meant to work as a database with a lot of
+writings. In case your repo's history gets very large, it is always possible to squash all the commits to get a
+fresh start.
+- **Number of operations per commit**: Once again, there is no hard-limit here. When a commit is done on the Hub, each
+git operation (addition or delete) is checked by the server before actually doing it. This means that when a hundred LFS
+files are committed at once, the server must check if each file has been correctly uploaded -checking its existence
+and size-. A timeout of 60s is set on the request, meaning that if the process takes more time an error is raised
+client-side. However, it can happen (in rare cases) that even if the timeout is raised client-side, the process is still
+completed server-side. This can be checked manually by browsing the repo on the Hub. To prevent this timeout, we recommend
+to add around 50-100 files per commit.
+
+To summarize it quickly:
+- Reach out to us for large repos (TBs of data)
```

Comment:
I think you could make a table out of this and stick it at the beginning of the section right before "Technical limitations" so users can quickly scan it. It would also help to do something like:

| Limitation               | Recommendation              |
|--------------------------|-----------------------------|
| Large repo (TBs of data) | contact us                  |
| Max 1M files/repo        | merge data into fewer files |
| Max 10k files/folder     | use subdirectories in repo  |

Reply:
Great idea!